hooks/tk-multi-publish2/basic/publish_session.py

```diff
@@ -374,6 +395,191 @@ def finalize(self, settings, item):
+
+        return ui_settings
+
+    def set_ui_settings(self, widget, settings, items=None):
+        """
+        This method is required to be defined in order for the custom UI to show up in the app.
+
+        Allows the custom UI to populate its fields with the settings from the
+        currently selected tasks.
+
+        The widget is the widget created and returned by
+        `create_settings_widget`.
+
+        A list of settings dictionaries are supplied representing the current
+        values of the settings for selected tasks. The settings dictionaries
+        correspond to the dictionaries returned by the settings property of the
+        hook.
+
+        Example::
+
+            settings = [
+            {
+                 "seeting_a": "/path/to/a/file"
+                 "setting_b": False
+            },
+            {
+                 "setting_a": "/path/to/a/file"
+                 "setting_b": False
+            }]
+
+        The default values for the settings will be the ones specified in the
+        environment file. Each task has its own copy of the settings.
+
+        When invoked with multiple settings dictionaries, it is the
+        responsibility of the custom UI to decide how to display the
+        information. If you do not wish to implement the editing of multiple
+        tasks at the same time, you can raise a ``NotImplementedError`` when
+        there is more than one item in the list and the publisher will inform
+        the user than only one task of that type can be edited at a time.
+
+        :param widget: The widget that was created by `create_settings_widget`.
+        :param settings: a list of dictionaries of settings for each selected
+            task.
+        :param items: A list of PublishItems the selected publish tasks are parented to.
+        """
+
+        if not settings:
+            return
+
+        if len(settings) > 1:
+            raise NotImplementedError
+
+        publish_mode_combobox = widget.property("publish_mode_combobox")
+        if not publish_mode_combobox:
+            self.logger.debug(
+                "Failed to retrieve Publish Mode combobox to set custom UI"
+            )
+            return
+
+        # Get the default setting for publish mode
+        default_value = (
+            self.settings.get("Publish Mode", {}).get("default")
+            or self.PUBLISH_MODE_DEFAULT
+        )
+
+        # Get the publish mode value from the settings, and set the combobox accordingly
+        publish_mode = settings[0].get("Publish Mode", default_value)
+        try:
+            publish_mode_index = max(self.PUBLISH_MODE_OPTIONS.index(publish_mode), 0)
+        except ValueError:
+            publish_mode_index = 0
+
+        # Set the publish mode combobox
+        current_version_index = publish_mode_combobox.currentIndex()
+        if current_version_index != publish_mode_index:
+            publish_mode_combobox.setCurrentIndex(publish_mode_index)
```

Comment:
Minor point: I would reverse the condition to keep the same function logic until the end (fail first)
```suggestion
        if current_version_index == publish_mode_index:
            # Nothing to do
            return

        publish_mode_combobox.setCurrentIndex(publish_mode_index)
```


But this is mostly a personal preference.